### PR TITLE
Updates to Token Modal 

### PR DESCRIPTION
**Pull request by ianlapham**
See #382 

#### Added
-  usd price fetching using sdk
- allBalances context that retrieves token balances for all supported (and usd market details too)
- updated styles on modal 
- fixes on mobile modal 
- text edits on advanced details (not modal related)

### CODE REVIEW

**Comment by NoahZinsmeister** [comment]
Wow looks sick!

DIP is slightly too far left because of the missing token.

Token amounts should be formatted with `formatSignificantDecimals` from the SDK (i think 6 sig figs?)

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (NoahZinsmeister) [inline .vscode/settings.json]: you should throw `.vscode/` in the `.gitignore`, or put it in your local global `.gitignore` (what i do)

File: public/locales/en.json [hunk 2]
```diff
@@ -36,6 +37,7 @@
   "youWillReceive": "You will receive at least",
   "youAreBuying": "You are buying",
   "itWillCost": "It will cost at most",
+  "forAtMost": "for at most",
```
Comment (NoahZinsmeister) [inline public/locales/en.json]: can't comment on the actual file, but again let's try to use feather icons wherever possible instead of rolling our own

i'd say `close.png` -> https://feathericons.com/?query=x

File: src/components/CurrencyInputPanel/index.js [hunk 3]
```diff
@@ -46,14 +45,17 @@ const InputRow = styled.div`
 `
 
 const Input = styled(BorderlessInput)`
-  font-size: 1.5rem;
+  font-size: 1rem;
```
Comment (NoahZinsmeister) [inline src/components/CurrencyInputPanel/index.js]: is this intentional? i think we should probably revert

File: src/components/ExchangePage/index.jsx [hunk 4]
```diff
@@ -116,13 +119,13 @@ function calculateEtherTokenInputFromOutput(outputAmount, inputReserve, outputRe
...
 
-function getInitialSwapState(outputCurrency) {
+function getInitialSwapState(state) {
   return {
     independentValue: '', // this is a user input
     dependentValue: '', // this is a calculated number
     independentField: INPUT,
-    inputCurrency: 'ETH',
-    outputCurrency: outputCurrency ? outputCurrency : ''
+    inputCurrency: state.inputCurrencyURL ? state.inputCurrencyURL : 'ETH',
```
Comment (NoahZinsmeister) [inline src/components/ExchangePage/index.jsx]: did this sneak in from the url stuff? should probably not include in this pr

File: src/components/ExchangePage/index.jsx [hunk 5]
```diff
@@ -232,7 +235,7 @@ function getMarketRate(
   }
 }
 
-export default function ExchangePage({ initialCurrency, sending }) {
+export default function ExchangePage({ initialCurrency, sending, inputCurrencyURL }) {
```
Comment (NoahZinsmeister) [inline src/components/ExchangePage/index.jsx]: same re. url stuff

File: src/hooks/index.js [hunk 6]
```diff
@@ -157,3 +157,5 @@ export function useCopyClipboard(timeout = 500) {
 
   return [isCopied, staticCopy]
 }
+
+export function useUsdPrice() {}
```
Comment (NoahZinsmeister) [inline src/hooks/index.js]: junk code? prob. delete

File: src/pages/Swap/index.js [hunk 7]
```diff
@@ -1,6 +1,16 @@
 import React from 'react'
 import ExchangePage from '../../components/ExchangePage'
+import { isAddress } from '../../utils'
 
-export default function Swap({ initialCurrency }) {
-  return <ExchangePage initialCurrency={initialCurrency} />
+function getQueryParam(windowLocation, name) {
```
Comment (NoahZinsmeister) [inline src/pages/Swap/index.js]: again, url stuff should prob go in another pr

Comment (ianlapham) [reply src/pages/Swap/index.js]: yep all this stuff snuck it - removing it for this PR

File: src/hooks/index.js (hunk 6 above)
Comment (ianlapham) [reply src/hooks/index.js]: Yep

**Comment by ianlapham** [comment]
> overall great job!
> 
> beyond the specific changes requested in the comments:
> 
> 1. didn't review the formatting code yet, but it looks good so i don't expect there to be many changes on that front
> 2. didn't review the allbalances context yet, or verify 100% that the usdprice context does what we want it to do, so i will do that soon (and probably push any changes straight to your branch)
> 3. how confident are you in the robustness of the sorting algo? only did a cursory overview, might not need to do more if you think it's good

Pretty confident in the new sorting. Its just one additional case added to compare balances. It does the comparison based on usd value (we do used value and not eth value because we only have the token balance and then the usd price of the token - could change this all to be based on eth value if there's any reason we'd be using that often in the future )
